RDFConsumer creates unnecessary second Context

**Describe the bug**
If you activate the RDFConsumer in DSpace 6 and finish a submission, you run into an Internal Server Error. The log shows:

```
org.hibernate.exception.GenericJDBCException: could not execute statement
...
Caused by: org.postgresql.util.PSQLException: ERROR: current transaction is aborted, commands ignored until end of transaction block
...
Caused by: org.postgresql.util.PSQLException: ERROR: duplicate key value violates unique constraint "doi_doi_key"
  Detail: Key (doi)=(10.7968/tlc.wi-test.0003/20) already exists.
```

The RDFConsumer creates a new Context in `READ_ONLY` mode and aborts that context later. Unfortunately another running context seem to be aborted as well. The easy fix here is to change RDFConsumer, to store the previous `Context.Mode`, set it to `READ_ONLY` and restore it once its work is done. The more complex solution would be to determine why both instances of `org.dspace.core.Context` are being aborted.

I don't know if the same bug occurs in DSpace 7, but the code exists there as well.

**Expected behavior**
Being able to submit an item when the RDFConsumer is activated (added to event.dispatcher.default.consumers).

**Related work**
PR is in preparation.

--- a/dspace-api/src/main/java/org/dspace/rdf/RDFConsumer.java
+++ b/dspace-api/src/main/java/org/dspace/rdf/RDFConsumer.java
@@ -286,51 +286,54 @@ public class RDFConsumer implements Consumer {
     @Override
     public void end(Context ctx) throws Exception {
         log.debug("Started processing of queued events.");
-        // create a new context, to be sure to work as anonymous user
-        // we don't want to store private data in a triplestore with public
-        // SPARQL endpoint.
-        ctx = new Context(Context.Mode.READ_ONLY);
-        if (toDelete == null) {
-            log.debug("Deletion queue does not exists, creating empty queue.");
-            this.toDelete = new LinkedList<>();
-        }
-        if (toConvert != null) {
-            log.debug("Starting conversion of DSpaceObjects.");
+        // store the context mode, set context read only for performance reasons, and restore the old mode
+        Context.Mode oldMode = ctx.getCurrentMode();
+        try {
+            ctx.setMode(Context.Mode.READ_ONLY);
+            if (toDelete == null) {
+                log.debug("Deletion queue does not exists, creating empty queue.");
+                this.toDelete = new LinkedList<>();
+            }
+            if (toConvert != null) {
+                log.debug("Starting conversion of DSpaceObjects.");
+                while (true) {
+                    DSOIdentifier id;
+                    try {
+                        id = toConvert.removeFirst();
+                    } catch (NoSuchElementException ex) {
+                        break;
+                    }
+
+                    if (toDelete.contains(id)) {
+                        log.debug("Skipping " + Constants.typeText[id.type] + " "
+                                      + id.id.toString() + " as it is marked for "
+                                      + "deletion as well.");
+                        continue;
+                    }
+                    log.debug("Converting " + Constants.typeText[id.type] + " "
+                                  + id.id.toString() + ".");
+                    convert(ctx, id);
+                }
+                log.debug("Conversion ended.");
+            }
+            log.debug("Starting to delete data from the triple store...");
             while (true) {
                 DSOIdentifier id;
                 try {
-                    id = toConvert.removeFirst();
+                    id = toDelete.removeFirst();
                 } catch (NoSuchElementException ex) {
                     break;
                 }
 
-                if (toDelete.contains(id)) {
-                    log.debug("Skipping " + Constants.typeText[id.type] + " "
-                                  + id.id.toString() + " as it is marked for "
-                                  + "deletion as well.");
-                    continue;
-                }
-                log.debug("Converting " + Constants.typeText[id.type] + " "
+                log.debug("Going to delete data from " +
+                              Constants.typeText[id.type] + " "
                               + id.id.toString() + ".");
-                convert(ctx, id);
+                delete(ctx, id);
             }
-            log.debug("Conversion ended.");
+        } finally {
+            // restore context mode
+            ctx.setMode(oldMode);
         }
-        log.debug("Starting to delete data from the triple store...");
-        while (true) {
-            DSOIdentifier id;
-            try {
-                id = toDelete.removeFirst();
-            } catch (NoSuchElementException ex) {
-                break;
-            }
-
-            log.debug("Going to delete data from " +
-                          Constants.typeText[id.type] + " "
-                          + id.id.toString() + ".");
-            delete(ctx, id);
-        }
-        ctx.abort();
         log.debug("Deletion finished.");
     }
 

